site-src/geps/gep-922.md

```diff
@@ -0,0 +1,177 @@
+
+### 1. API Versions (ex: v1alpha2)
+Each API version provides a unique way to interact with the API. Significant
+changes such as removing or renaming fields will be represented here.
+
+### 2. Bundle Versions (ex: v0.4.0)
+Bundle versions are broader in scope. They use semantic versioning and include
+the following:
+
+* API Types/CRDs
+* Validating Webhook
+
+Each bundle may include multiple API versions, potentially introducing new ones
+and/or removing old ones as part of a new bundle.
+
+## Limitations of Webhook and CRD Validation
+CRD and webhook validation is not the final validation i.e. webhook is “nice UX”
+but not schema enforcement. This validation is intended to provide immediate
+feedback to users when they provide an invalid configuration, but can not
+completely be relied on because it:
+
+* Is not guaranteed to be present or up to date in all clusters.
+* Will likely never be sufficient to cover all edge cases.
+* May be loosened in future API releases.
+
+## Persona Requirements
+When implementing or using Gateway API, each persona has a unique set of
+responsibilities to ensure we're providing a consistent experience.
+
+### API Authors:
+* MUST provide conversion between stable API versions, starting with v1alpha2.
+* MAY include the following changes to an existing API version with a new bundle
+ **patch** version:
+    * Clarifications to godocs
+    * Loosening validation to fix a bug (not to enable a new feature)
+    * Fixes to typos
+* MAY include the following changes to an existing API version with a new bundle
+  **minor** version:
+    * Everything that is valid in a patch release
+    * New experimental API fields or resources
+    * Loosened validation
+    * Making required fields optional
+    * Removal of experimental fields
+    * Removal of experimental resources
+    * Graduation of fields or resources from experimental to stable track
+* MAY introduce a new **API version** with a new bundle minor version, which may
+  include:
+    * Everything that is valid in a minor release
+    * Renamed fields
+    * Anything else that is valid in a new Kubernetes API version
+    * Removal/tombstoning of beta fields
+* MAY release a new major bundle version (v1.0) as part of graduating the API to
+  GA and releasing a new API version.
+
+Note that each new bundle version, no matter how small, may include updated
+CRDs, webhook, or both. Implementations may read annotations on Gateway API CRDs
+(defined below) to determine the version and channel of CRDs that have been
+installed in the cluster.
+
+### Implementers:
+* MUST handle fields with loosened validation without crashing
+* MUST handle fields that have transitioned from required to optional without
+  crashing
+* MUST NOT rely on webhook or CRD validation as a security mechanism. If field
+  values need to be escaped to secure an implementation, both webhook and CRD
+  validation can be bypassed and cannot be relied on. Instead, implementations
```

Comment:
This wording is fine for now, but if we do implement the validation module, feels like it should be a MUST to me.

Reply:
Good catch, I added a note to https://github.com/kubernetes-sigs/gateway-api/issues/926 so we remember to come back to this GEP and update it.